### ISSUE TITLE
Possible bug in ComputePipelineCreateInfo.PopulateCPointer()?

### ISSUE THREAD

**Issue by kgigitdev** [vkngwrapper/core#5]
Hi there.

I've been experimenting with trying to get a minimal Vulkan compute example up using vkngwrapper, and I've hit a brick wall with what _might_ be a bug in the library.

However, I basically know next to nothing about Vulkan, so I apologise profusely in advance if this is not the case.

I've been following [this](https://github.com/necrashter/minimal-vulkan-compute-shader) as my main reference, which is mostly a C++ified version of [this](https://www.duskborn.com/posts/a-simple-vulkan-compute-example/).

Specifically, the error I get is when calling `core1_0.Device.CreateComputePipelines()`. All the examples I have found seem to imply that `ComputePipelineCreateInfo.BasePipeline` may be set to null. For example, in the C example from the latter of the two examples above, we see:

```
VkComputePipelineCreateInfo computePipelineCreateInfo = {
  VK_STRUCTURE_TYPE_COMPUTE_PIPELINE_CREATE_INFO,
  0, 0,
  {
    VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO,
    0, 0, VK_SHADER_STAGE_COMPUTE_BIT, shader_module, "f", 0
  },
  pipelineLayout, 0, 0
};
```

in which the final two arguments are the base pipeline handle and the number of base pipelines (therefore, a null handle and zero, respectively).

However, in `pipeline_createinfo.go` line 248, there is:

```
createInfo.basePipelineHandle = C.VkPipeline(unsafe.Pointer(o.BasePipeline.Handle()))
createInfo.basePipelineIndex = C.int32_t(o.BasePipelineIndex)
```

In other words, an unconditional attempt to dereference the `BasePipeline` field.

There is an analogous field for the graphic pipeline, and indeed, in the examples, this is treated thus in `examples/lunarg_samples/draw_subpasses/main.go`:

```
        pipelineOptions := core1_0.GraphicsPipelineCreateInfo{
                Layout:            info.PipelineLayout,
                BasePipeline:      nil,
                BasePipelineIndex: 0,
...
```

However, the creation code is different for the graphics pipeline, which just does this:

```
createInfo.basePipelineHandle = (C.VkPipeline)(nil)
createInfo.basePipelineIndex = C.int32_t(o.BasePipelineIndex)
```

This seems reasonable; according to [this](https://stackoverflow.com/questions/37135130/vulkan-creating-and-benefit-of-pipeline-derivatives), derived pipelines seem to be out of favour anyway, so not supporting them is no great loss.

Is this a bug? Or am I missing something very obvious?

Apart from this minor problem, I'd like to take this opportunity to thank you for the library. It has generally been a pleasure to use, and is remarkably go-idiomatic.

One minor thing I wonder is whether there's an easier way of being able to determine the available flags for a given `Flag` field, without having to go to the Vulkan documentation and then search for likely symbols. I use goland, so I can jump to the definition of the flag type, but the flag constants are not defined anywhere nearby in the source code.

**Comment by CannibalVox** [comment]
I think you're right that it's a bug. I will put together a fix.

As far as flags go, I had two choices when putting them together:

1) Put all flags for a type in the same place, where the type is defined. This would mean that we keep core1_0 up to date with all the extension-level and core1.1+ flags that are later added to flag types, but it would be easy to locate. downside upside here is that it's very easy to make a mistake by attempting to use a flag that is not available in your current environment because you are confused about where and why it is available in the heat of the moment. It also increases the amount of old core version updates that need to happen as a result of new extensions or core versions.
2) Put all flags in the place where they are defined.  So harder to find out what flags exist for a given flag type, but avoids the problems with the above.

My thought process is that the specialist nature of vulkan means that it's rare to go "shopping" for flags as you would in other APIs. You generally have a purpose & intention when creating a structure. That may end up being inaccurate, though.

**Comment by CannibalVox** [comment]
If it helps at all, the godoc comments for the flag type contain a link to the vulkan spec, you should be able to ctrl+click it to get a list of flags

**Comment by CannibalVox** [comment]
This should now be fixed- please let me know if v2.0.1 works better for you

**Comment by kgigitdev** [comment]
Hi,

Yes, I can confirm that the problem I was having is fixed by v2.0.1. Many thanks for the quick turnaround, especially at a time of year when exactly the opposite would be expected!

Now to see if can reach the point where I can get the GPU to actually compute anything ...

**Comment by CannibalVox** [comment]
Great to hear!
